### DISCUSSION TITLE
Kool math

### DISCUSSION THREAD

**Discussion by antzGames** [fabmax/kool#52] [General]
Hello all.  First of all I want to say what you done here is amazing.

My question is on the kool math classes.  I did not see Quaternion.  I am wondering if any of you using Kool have encountered any gimbal lock issues specifically when rotating 3D objects or the camera?

**Comment by fabmax** [comment]
There is no dedicated quaternion class. However, `(Mutable)Vec4` has a few quaternion / rotation functions and `Mat3` and `Mat4` have corresponding `setRotate(Vec4)` functions. It's not the nicest api but it should do the trick.

**Comment by antzGames** [comment]
Yes I see them now.  Thanks.
